components/logins/src/logins.udl

```diff
@@ -0,0 +1,116 @@
+namespace logins {
+
+    [Throws=LoginsStorageError]
+    string open_and_get_salt(string path, [ByRef] string encryption_key);
+
+    [Throws=LoginsStorageError]
+    void open_and_migrate_to_plaintext_header(string path, [ByRef] string encryption_key, [ByRef] string salt);
+};
+
+
+// Converted from mozilla.appservices.logins.protobuf.rs
+dictionary LoginRecord {
+    string guid;
+    string hostname;
+    string password;
+    string username;
+    string? http_realm;
+    string? form_submit_url;
+    string username_field;
+    string password_field;
+    i64 times_used;
+    i64 time_created;
+    i64 time_last_used;
+    i64 time_password_changed;
+};
+
+dictionary MigrationMetrics {
+    MigrationPhaseMetrics fixup_phase;
+    MigrationPhaseMetrics insert_phase;
+    u64 num_processed;
+    u64 num_succeeded;
+    u64 num_failed;
+    u64 total_duration;
+    sequence<string> errors;
+};
+
+dictionary MigrationPhaseMetrics {
+    u64 num_processed;
+    u64 num_succeeded;
+    u64 num_failed;
+    u64 total_duration;
+    sequence<string> errors;
+};
+
+[Error]
+enum LoginsStorageError {
+    "UnexpectedLoginsStorageError",
+    "SyncAuthInvalid",
+    "MismatchedLock",
+    "NoSuchRecord",
+    "IdCollision",
+    "InvalidRecord",
+    "InvalidKey",
+    "RequestFailed",
+    "Interrupted",
+};
+
+enum InvalidLoginReason {
+    "EmptyOrigin",
+    "EmptyPassword",
+    "DuplicateLogin",
+    "BothTargets",
+    "NoTarget",
+    "IllegalFieldValue",
+};
+
+interface PasswordStore {
```

Comment:
I agree that LoginStore name would suit this better. Unfortunately [db.rs](https://github.com/mozilla/application-services/blob/main/components/logins/src/db.rs#L1228) already has a LoginStore. Two things we can do here:

1. Rename the db.rs something like SyncLoginStore, since it seems specifically for the SyncEngine
2. Completely remove the db.rs one and add the `scope` var to the PasswordStore.

Reply:
I would lean towards (1) just because it'll be the safest option during this initial round of work

Reply:
I agree re (1) - maybe `RustLoginStore` is a better name? I actually owe you a PR for the sync manager work, so leave that with me.